src/utils/utils.ts

```diff
@@ -62,7 +64,93 @@ function standardErrorReplacer(key: string, value: any) {
   return value;
 }
 
-function outputFormatter(msg: OutputObject): string | Uint8Array {
+async function* arrayToAsyncIterable<T>(array: T[]): AsyncIterable<T> {
+  for (const item of array) {
+    yield item;
+  }
+}
+
+// Function to handle 'table' type output
+const outputTableFormatter = async (
+  rowStream: TableRow[] | AsyncIterable<TableRow>,
+  options?: TableOptions,
+): Promise<string> => {
+  let output = '';
+  const maxColumnLengths: Record<string, number> = {};
+  let rowCount = 0;
+
+  // Initialize maxColumnLengths with header lengths if headers are provided
+  if (options?.headers) {
+    for (const header of options.headers) {
+      maxColumnLengths[header] = header.length;
+    }
+  }
+
+  let iterableStream = Array.isArray(rowStream)
+    ? arrayToAsyncIterable(rowStream)
+    : rowStream;
```

Comment:
Why is this needed? It's perfectly valid to do 

```ts
const array = [1, 2, 3];

for await (const value of array) {
  //do things
}

```

Remember, you can `await` anything, not just promises.

You don't need to convert an array to an async iterator. Unless you want to use iterator calls like `thing.next()`?

Reply:
Hey @tegefaulkes, so using `for await (const value of array)` is actually worse in terms of space complexity as you have to store the array instead of just reading it, both your approach and mine use the same time complexity of O(2n), which is what Roger wanted, so I don't see whats wrong with my implementation here. Please do elaborate.

Reply:
I think you misunderstand, I'm saying you can consume an array and an `AsyncIterable` the same way. You don't need to convert an array into an `AsyncIterable` to consume it with a async for of loop.

Reply:
> Hey @tegefaulkes, so using `for await (const value of array)` is actually worse in terms of space complexity as you have to store the array instead of just reading it, both your approach and mine use the same time complexity of O(2n), which is what Roger wanted, so I don't see whats wrong with my implementation here. Please do elaborate.

i don't think there is any performance difference between

```
const array = [1, 2, 3];

for await (const value of array) {
  //do things
}
```

and

```
const array = [1, 2, 3];

for (const value of array) {
  //do things
}
```

Also, `arrayToAsyncIterable` would create a AsyncIterable instance, where as  `for (const value of array)` wouldn't need to